pkg/commands/server/plugins.go

```diff
@@ -62,3 +63,37 @@ func cmdPlugins(c *cli.Context) error {
 	}
 	return formatter.Write()
 }
+
+// getPlugins is a helper function that takes a Context and an arbitrary number
+// of string tags as arguments and returns a set of matched plugins.
+func getPlugins(c *cli.Context, tags ...string) ([]scheme.Plugin, error) {
+	var plugins []scheme.Plugin
+
+	pluginsResults, err := client.Client.Plugins()
+	if err != nil {
+		return nil, err
+	}
+
+	// counter keeps count of empty tag string. If all tags are empty, meaning
+	// that no arguments are provided, the function returns all configured plugins.
+	// Otherwise, it only returns these matched ones.
+	counter := 0
+	for _, tag := range tags {
+		if tag == "" {
+			counter++
+			continue
+		}
+
+		for _, plugin := range pluginsResults {
+			if tag == plugin.Tag {
+				plugins = append(plugins, plugin)
+			}
+		}
+	}
```

Comment:
this seems like a strange way of doing this, mostly because I'm not sure that an empty string tag could be passed in via command line. If no tag strings are passed in, then `tags` should be empty, so I think it would probably good enough to just do something like:

```go
if len(tags) == 0 {
	return pluginResults, nil
}

for _, tag := range tags {
	for _, plugin := range pluginsResults {
		if tag == plugin.Tag {
			plugins = append(plugins, plugin)
			break
		}
	}
}

return plugins, nil
```

Reply:
It should be updated by now.